File: Tests/Base/BaseTestCase.m
//
//  BaseTestCase.m
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 2/4/16.
//  Copyright © 2016 anton. All rights reserved.
//

#import "BaseTestCase.h"

@implementation BaseTestCase

///--------------------------------------
#pragma mark - Helpers
///--------------------------------------

- (XCTestExpectation *)currentSelectorTestExpectation {
	NSInvocation *invocation = self.invocation;
	NSString *selectorName = invocation ? NSStringFromSelector(invocation.selector) : @"testExpectation";
	return [self expectationWithDescription:selectorName];
}

- (void)waitForTestExpectations {
	[self waitForExpectationsWithTimeout:10.0 handler:nil];
}

@end

File: Tests/Base/BaseTestCase.swift
//
//  BaseTestCase.swift
//  QBRTCDemo
//
//  Created by Anton Sokolchenko on 05.12.16.
//  Copyright © 2016 anton. All rights reserved.
//

import Foundation
import XCTest

class BaseTestCase: XCTestCase {
	func segues(ofViewController viewController: UIViewController) -> [String] {
		let identifiers = (viewController.value(forKey: "storyboardSegueTemplates") as? [AnyObject])?.compactMap({ $0.value(forKey: "identifier") as? String }) ?? []
		return identifiers
	}
	
	func waitForTimeInterval(_ milliseconds: UInt64) {
		var waitCondition = true
		DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + Double(Int64(milliseconds * NSEC_PER_MSEC)) / Double(NSEC_PER_SEC)) {
			waitCondition = false
		}
		while(waitCondition) { RunLoop.current.run(mode: RunLoop.Mode.default, before: Date.distantFuture) }
	}
}
